Assets/js/05.budget_widget.js

```diff
@@ -0,0 +1,99 @@
+                        autoFill: true,
+                        columns: response.columns,
+                        order: [[2, 'asc'], [4, 'asc']],
+                        bLengthChange: false,
+                        lengthMenu: [[rowCount]],
+                        dom: '<<lBf>rtip>',
+                        buttons: [
+                            'excelHtml5',
+                            'csvHtml5'
+                        ],
+                        columnDefs: [
+                            {
+                                render: function (data, type, row) {
+                                    return renderStatusToggle(row[0]);
+                                },
+                                targets: 0
+                            },
+                            {
+                                // campaign name and link
+                                render: function (data, type, row) {
+                                    return renderLink(row[1], row[2]);
+                                },
+                                targets: 1
+                            },
+                            {
+                                // source name and link
+                                render: function (data, type, row) {
+                                    return renderLink(row[3], row[4]);
+                                },
+                                targets: 3
+                            },
+                            {
+                                // forecast
+                                render: function (data, type, row) {
+                                    return renderForecast(row[8], row[9]);
+                                },
+                                targets: 8
+                            },
+                            {visible: false, targets: [2, 4, 9]},
+                            {width: '5%', targets: [0]},
+                            {width: '45%', targets: [5]},
+                        ]
+                    });
+                    mQuery('#budget-widget-overlay').hide();
+                    mQuery('#budgets-widget_wrapper .dt-buttons').css({float: "right", marginLeft: "10px"});
+                }
+            });
+        });
+    }
+};
+
+function renderStatusToggle (icon) {
+    var color = '#cb4641';
+    if (icon == 'fa-heartbeat') {
+        color = '#2a84c5';
+    }
+    return '<i style="color:' + color + ';" class="fa ' + icon + '"></i>';
+};
+
+function renderLink (label, link) {
+    return '<a href="' + link + '">' + label + '</a>';
+};
+
+function renderForecast(value, forecast) {
+    // danger or success
+    var forecastClass = 'success';
+    var forecastInt = parseInt(forecast);
+    if(forecastInt >=90){
+        forecastClass = 'danger';
+    }
+    return '<span class="label label-' + forecastClass + '">' + value + '</span>&nbsp;<span class="label label-' + forecastClass + '">' + forecast + '</span>';
+}
+
+mQuery(document).ready(function () {
```

Comment:
Didn't we discover that we shouldn't be using document.ready()?

Reply:
yes. didnt seem to hurt anything, but the whole things seems unnecessary now.